shell/browser/relauncher.cc

```diff
@@ -57,6 +57,11 @@ bool RelaunchAppWithHelper(const base::FilePath& helper,
   StringVector relaunch_argv;
   relaunch_argv.push_back(helper.value());
   relaunch_argv.push_back(internal::kRelauncherTypeArg);
+  // Relauncher process has its own --type=relauncher which
+  // is not recognized by the service_manager, explicitly set
+  // the sandbox type to avoid CHECK failure in
+  // service_manager::SandboxTypeFromCommandLine
+  relaunch_argv.push_back("--no-sandbox");
```

Comment:
Is this definitely not propagated to the new process? It would suck if `app.relaunch()` ===> no sandbox any more

Reply:
`internal::kRelauncherArgSeparator` makes sure what sticks with the relauncher process and what gets passed to the app https://github.com/electron/electron/blob/master/shell/browser/relauncher.cc#L156-L167

Reply:
```
 4018 ?        Ssl    0:03 /snap/code-insiders/429/usr/share/code-insiders/code-insiders --no-sandbox electron-quick-start/
 4989 pts/0    Sl+    0:00 /home/parallels/github/electron-gn/src/out/Testing/electron --enable-logging ./electron-quick-start/
 4992 pts/0    S+     0:00 /home/parallels/github/electron-gn/src/out/Testing/electron --type=zygote --no-zygote-sandbox --enable-logging
 4993 pts/0    S+     0:00 /home/parallels/github/electron-gn/src/out/Testing/electron --type=zygote --enable-logging
 4995 pts/0    S+     0:00 /home/parallels/github/electron-gn/src/out/Testing/electron --type=zygote --enable-logging
 5021 pts/0    Sl+    0:00 /home/parallels/github/electron-gn/src/out/Testing/electron --type=gpu-process --field-trial-handle=14602157809997563534,390724223982036700,131072 --enable-features=WebComponentsV0Enabled --disable-features=SpareRendererForSitePerProcess --enable-logging --gpu-preferences=MAAAAAAAAAAgAAAAAAAAAAAAAAAAAAAAAABgAAAAAAAQAAAAAAAAAAAAAAAAAAAACAAAAAAAAAA= --enable-logging --shared-files
 5023 pts/0    Sl+    0:00 /home/parallels/github/electron-gn/src/out/Testing/electron --type=utility --field-trial-handle=14602157809997563534,390724223982036700,131072 --enable-features=WebComponentsV0Enabled --disable-features=SpareRendererForSitePerProcess --lang=en-US --service-sandbox-type=network --enable-logging --enable-logging --shared-files=v8_snapshot_data:100
 5024 pts/0    S+     0:00 /home/parallels/github/electron-gn/src/out/Testing/electron --type=broker
 5031 pts/0    S+     0:00 /home/parallels/github/electron-gn/src/out/Testing/electron --type=relauncher --no-sandbox --- --enable-logging --second /home/parallels/github/electron-gn/src/out/Testing/electron ./electron-quick-start/
 5039 pts/0    Sl+    0:00 /home/parallels/github/electron-gn/src/out/Testing/electron --type=renderer --enable-logging --file-url-path-alias=/gen=/home/parallels/github/electron-gn/src/out/Testing/gen --field-trial-handle=14602157809997563534,390724223982036700,131072 --enable-features=WebComponentsV0Enabled --disable-features=SpareRendererForSitePerProcess --lang=en-US --app-path=/home/parallels/github/electron-quick-start --enable-sandbox --native-window-open --background-color=#fff --enable-spellcheck --disable-electron-site-instance-overrides --num-raster-threads=2 --enable-main-frame-before-activation --renderer-client-id=4 --no-v8-untrusted-code-mitigations --shared-files=v8_snapshot_data:100
 5064 pts/1    S+     0:00 grep --color=auto electron

```